compiler-rt/test/rtsan/deduplicate_errors.cpp

```diff
@@ -0,0 +1,37 @@
+// RUN: %clangxx -fsanitize=realtime %s -o %t
+// RUN: env RTSAN_OPTIONS="halt_on_error=false,print_stats_on_exit=true" %run %t 2>&1 | FileCheck %s
+// RUN: env RTSAN_OPTIONS="halt_on_error=false" %run %t 2>&1 | grep "==ERROR: RealtimeSanitizer:" | wc -l | awk '{exit $1 != 4}'
```

Comment:
Wanted to draw people's attention here (especially @davidtrevelyan). This is a pattern I see in other lit tests and is a nice way to count occurrences of a thing

Reply:
There has been an effort about migrating tests to the lit-internal shell (CC @ilovepi) which would not support this. So I think from that point of view four `CHECK` followed by `CHECK-NOT` would be better

Reply:
Glad I pointed this out!! I'll fix it up

Reply:
I believe CHECK-COUNT is the way you normally do this: https://llvm.org/docs/CommandGuide/FileCheck.html#the-check-count-directive, but I don’t have a strong preference on that.  In either case the check seems brittle.  I don’t have a better solution off the top of my head though.

Reply:
Also I’m not 100% on the awl but causing an issue for the internal shell, but I think it’s always preferred if we can do the same thing with FileCheck.

Reply:
how would awk work? it depends on the host system?

Reply:
> how would awk work? it depends on the host system?

Yeah. we also don't require it in our docs, so 🤷  IDK what level of support we expect for that. Likely, many tests w/ that pattern are `REQUIRES: shell` or `UNSUPPORTED: windows`. We list the utilities you're expected to have for testing/building here: https://llvm.org/docs/GettingStarted.html#software

Reply:
Thanks for the advice, I didn't realize CHECK-COUNT existed. Updated the test to reflect that.